src/components/Company/FederalTaxes/Form.tsx

```diff
@@ -0,0 +1,103 @@
+import { useTranslation, Trans } from 'react-i18next'
+import { useFormContext } from 'react-hook-form'
+import { ListBoxItem, Link } from 'react-aria-components'
+import {
+  TaxPayerType,
+  FilingForm,
+} from '@gusto/embedded-api/models/operations/putv1companiescompanyidfederaltaxdetails'
+import type { FederalTaxFormInputs } from './FederalTaxes'
+import { useFederalTaxes } from './FederalTaxes'
+import { TextField, Select, Flex } from '@/components/Common'
+import { usePlaceholderEin, normalizeEin } from '@/helpers/federalEin'
+
+export function Form() {
+  const { t } = useTranslation('Company.FederalTaxes')
+  const { federalTaxDetails } = useFederalTaxes()
+  const { control, setValue } = useFormContext<FederalTaxFormInputs>()
+
+  const placeholderEin = usePlaceholderEin(federalTaxDetails?.hasEin)
+
+  const taxPayerTypeOptions = Object.values(TaxPayerType).map(value => ({
```

Comment:
Is there optimization in memoizing them so they are not computed every time the component re-renders/diffs?